doc/Migration.md

```diff
@@ -0,0 +1,54 @@
+# Migration Guide
+
+## Description
+
+The following guide highlights potential migration steps necessary during `theia` upgrades discovered when adopting the framework.
+Please see the latest version (`master`) for the most up-to-date information.
+
+## Guide
+
+### v1.16.0
+
+[Release](https://github.com/eclipse-theia/theia/releases/tag/v1.16.0)
+
+- N/A.
+
+### v1.15.0
+
+[Release](https://github.com/eclipse-theia/theia/releases/tag/v1.15.0)
+
+#### Keytar:
+
+- [`keytar`](https://github.com/atom/node-keytar) was added as a dependency for the secrets API. and may require `libsecret` in your particular distribution to be functional:
+  - Debian/Ubuntu: `sudo apt-get install libsecret-1-dev`
+  - Red Hat-based: `sudo yum install libsecret-devel`
+  - Arch Linux: `sudo pacman -S libsecret`
+- It is possible that a `yarn resolution` is necessary for `keytar` to work on older distributions (the fix was added in `1.16.0` by downgrading the dependency version):
```

Comment:
For rebuilding under Alpine, it requires `apk add libsecret-dev`.
We faced it in Che-Theia. So, I believe it's worth mentioning as well.